### PR TITLE
Add FileDescriptorGaugeSet

### PR DESCRIPTION
**Pull request by jhaals**
Add metric for the ratio of allocated file descriptors

### CODE REVIEW

**Comment by bogdad** [comment]
👍

**Comment by jhaals** [comment]
ffwd output running locally.

```
...
13:59:43.759 [ffwd-worker-1] INFO  com.spotify.ffwd.debug.DebugPluginSink - M: Metric(key=jvm-example.jvm-fd-ratio, value=0.0033203125, time=null, host=null, riemannTags=[], tags={unit=ratio, what=file-descriptor-ratio, metric_type=gauge}, proc=null)
...
```

**Comment by jhaals** [comment]
For the reviewer: I'm unsure if the `unit` "ratio" is correct/good, please let me know if you want it to be "count" instead.

File: core/src/main/java/com/spotify/metrics/jvm/FileDescriptorGaugeSet.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public FileDescriptorGaugeSet() {
+        fileDescriptorRatioGauge = new FileDescriptorRatioGauge();
+    }
+
+    @Override
+    public Map<MetricId, Metric> getMetrics() {
+        fileDescriptorRatioGauge.getValue();
+        final Map<MetricId, Metric> gauges = new HashMap<MetricId, Metric>();
+        final MetricId metricId =
+                MetricId.build().tagged("what", "file-descriptor-ratio", "unit", "ratio");
```
Comment (gabrielgerhardsson) [inline core/src/main/java/com/spotify/metrics/jvm/FileDescriptorGaugeSet.java]: As to your question about the unit: I believe that "%" would be reasonable. It's used in other places and fits well with a value in the range of [0, 1].

**Comment by gabrielgerhardsson** [comment]
@jhaals Thanks. Feel free to squash and merge 👍

**Comment by jhaals** [comment]
Thanks for reviewing @gabrielgerhardsson!
Guess someone in hero needs to 🔪  a new release :)

**Comment by gabrielgerhardsson** [comment]
@jhaals No worries! :)
I've created a card in our backlog to create a release.

**Pull request merged**